ai-server/app/api/v1/music_api.py

```
@@ -1,50 +1,50 @@
+            raise GenerationError("Generator returned no s3_key")
+
+        return GenerateMusicResponse(
+            status="success",
+            message="Music generated successfully",
+            data={"s3_key": result["s3_key"]},
         )
+
+    except AIServerError:
+        raise
+    except Exception as e:
+        raise AIServerError(f"Unexpected error during generation: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)